### PR TITLE
resource/aws_rds_cluster_instance: Ensure performance_insights_kms_key_id updates include performance_insights_enabled

### LINKED ISSUE #3015: resource/aws_rds_cluster_instance: Unable to modify performance insights key without updating the "enable parameter
When updating the KMS key used by performance insight on a cluster instance, the apply fails:

> InvalidParameterCombination: Can not set PerformanceInsightsKMSKeyId without EnablePerformanceInsights

This is due to changes in #2331, the code we currently have looks like this:

```
	if d.HasChange("performance_insights_enabled") {
		d.SetPartial("performance_insights_enabled")
 		req.EnablePerformanceInsights = aws.Bool(d.Get("performance_insights_enabled").(bool))
 		requestUpdate = true
 	}
 
 	if d.HasChange("performance_insights_kms_key_id") {
 		d.SetPartial("performance_insights_kms_key_id")
 		req.PerformanceInsightsKMSKeyId = aws.String(d.Get("performance_insights_kms_key_id").(string))
 		requestUpdate = true
 	}
 ```

Currently from the AWS perspective it makes no sense to have a partial `PerformanceInsightsKMSKeyId` without `EnablePerformanceInsights` being updated as well. Those two should be merged into the same condition.

cc @kwent, @Ninir

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #3015

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

This update ensures the correct RDS API error is shown when attempting to modify the Performance Insights KMS Key ID (we use `ExpectError` in the testing so this passes):

```
--- FAIL: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_DefaultKeyToCustomKey (691.76s)
    testing.go:568: Step 2 error: errors during apply:

        Error: Error modifying DB Instance tf-acc-test-2133090011528324987: InvalidParameterCombination: You cannot change your Performance Insights KMS key
```

Previously before code update:

```
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql2 (587.54s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql1 (594.85s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_CustomKey (660.70s)
--- FAIL: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_DefaultKeyToCustomKey (667.49s)
    testing.go:568: Step 2 error: errors during apply:

        Error: Error modifying DB Instance tf-acc-test-4757962708939555369: InvalidParameterCombination: Can not set PerformanceInsightsKMSKeyId without EnablePerformanceInsights
        	status code: 400, request id: c64b9f56-1b56-420c-9575-c5cfed87270b

          on /var/folders/v0/_d108fkx1pbbg4_sh864_7740000gn/T/tf-test833219777/main.tf line 15:
          (source code not available)

--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraPostgresql (699.46s)
```

Output from acceptance testing:

```
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql2_DefaultKeyToCustomKey (630.56s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql1 (649.52s)
--- PASS: TestAccAWSRDSClusterInstance_generatedName (651.77s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql1_DefaultKeyToCustomKey (652.94s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraPostgresql (670.18s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_CustomKey (671.61s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraPostgresql_DefaultKeyToCustomKey (672.19s)
--- PASS: TestAccAWSRDSClusterInstance_PubliclyAccessible (683.28s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql2 (690.78s)
--- PASS: TestAccAWSRDSClusterInstance_disappears (730.05s)
--- PASS: TestAccAWSRDSClusterInstance_CopyTagsToSnapshot (732.08s)
--- PASS: TestAccAWSRDSClusterInstance_az (747.28s)
--- PASS: TestAccAWSRDSClusterInstance_namePrefix (764.03s)
--- PASS: TestAccAWSRDSClusterInstance_kmsKey (792.98s)
--- PASS: TestAccAWSRDSClusterInstance_withInstanceEnhancedMonitor (822.65s)
--- PASS: TestAccAWSRDSClusterInstance_basic (1516.75s)
```

### CODE REVIEW

File: aws/resource_aws_rds_cluster_instance_test.go [hunk 1]
```diff
@@ -852,6 +1044,71 @@ resource "aws_db_parameter_group" "bar" {
 
 func testAccAWSClusterInstanceConfigPerformanceInsightsEnabledAuroraMysql1(rName string) string {
 	return fmt.Sprintf(`
+resource "aws_rds_cluster" "test" {
+  cluster_identifier  = %[1]q
+  database_name       = "mydb"
+  engine              = "aurora"
+  master_password     = "mustbeeightcharaters"
```
Comment (aeschright) [inline aws/resource_aws_rds_cluster_instance_test.go]: Not important, but --
```suggestion
  master_password     = "mustbeeightcharacters"
```

Comment (aeschright) [reply aws/resource_aws_rds_cluster_instance_test.go]: There's more of the same below, I assume from copy/paste. :)

**Comment by aeschright** [comment]
I confirm:

```
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql1 (638.20s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql1 (639.26s)
--- PASS: TestAccAWSRDSClusterInstance_namePrefix (657.66s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraMysql2 (658.93s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraPostgresql (659.28s)
--- PASS: TestAccAWSRDSClusterInstance_generatedName (669.38s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql2 (669.93s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql1_DefaultKeyToCustomKey (679.68s)
--- PASS: TestAccAWSRDSClusterInstance_CopyTagsToSnapshot (683.99s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsEnabled_AuroraPostgresql (709.28s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraMysql2_DefaultKeyToCustomKey (739.71s)
--- PASS: TestAccAWSRDSClusterInstance_withInstanceEnhancedMonitor (741.46s)
--- PASS: TestAccAWSRDSClusterInstance_PerformanceInsightsKmsKeyId_AuroraPostgresql_DefaultKeyToCustomKey (759.47s)
--- PASS: TestAccAWSRDSClusterInstance_az (787.42s)
--- PASS: TestAccAWSRDSClusterInstance_disappears (829.10s)
--- PASS: TestAccAWSRDSClusterInstance_PubliclyAccessible (855.10s)
--- PASS: TestAccAWSRDSClusterInstance_kmsKey (1252.82s)
--- PASS: TestAccAWSRDSClusterInstance_basic (1479.04s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1479.951s
```

File: aws/resource_aws_rds_cluster_instance_test.go (hunk 1 above)
Comment (bflad) [reply aws/resource_aws_rds_cluster_instance_test.go]: Good eyes 😎 fixed across the file 👍

**Pull request merged**